### ISSUE TITLE
【Question】No statistics info found

### ISSUE THREAD

**Issue by YannickXiong** [myzhan/boomer#17]
I tried your ideal, it was great.'
There's still a question, how to  get the statistical info  on performance metrics, like qps, response time?
When boomer is running, the statistics info on the GUI is none.

**Comment by myzhan** [comment]
You need to run Locust as a master, and make sure boomer connect to the master, then you can see infos on the Web-UI.

Or, maybe your boomer task have issues, please try boomer's examples instead of yours.

**Comment by YannickXiong** [comment]
Yeah, locust master meet some running errors.

xiongyangdeiMac:python3 xiongyang$ locust -f rfinex_locust.py --master --master-bind-host=127.0.0.1 --master-bind-port=18089
[2017-11-21 11:17:58,448] xiongyangdeiMac.local/INFO/locust.main: Starting web monitor at *:8089
[2017-11-21 11:17:58,542] xiongyangdeiMac.local/INFO/locust.main: Starting Locust 0.8
[2017-11-21 11:19:10,040] xiongyangdeiMac.local/INFO/locust.runners: Client 'xiongyangdeiMac.local_fb6d4a17712462241134a032a9355a89' reported as ready. Currently 1 clients ready to swarm.
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: Traceback (most recent call last):
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/gevent/greenlet.py", line 536, in run
    result = self._run(*self.args, **self.kwargs)
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/runners.py", line 332, in client_listener
    events.slave_report.fire(client_id=msg.node_id, data=msg.data)
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/events.py", line 27, in fire
    handler(**kwargs)
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,057] xiongyangdeiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/stats.py", line 586, in on_slave_report
    global_stats.total.extend(StatsEntry.unserialize(data["stats_total"]))
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: KeyError: 'stats_total'
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: Tue Nov 21 11:19:13 2017
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: 
[2017-11-21 11:19:13,058] xiongyangdeiMac.local/ERROR/stderr: <Greenlet at 0x10ce699c8: <bound method MasterLocustRunner.client_listener of <locust.runners.MasterLocustRunner object at 0x10ce9c8d0>>> failed with KeyError

**Comment by myzhan** [comment]
Thank you for reporting, it is a bug. Because locust have changed the data protocol.

It's fixed. Please try the last version of boomer.

go get -u github.com/myzhan/boomer

**Comment by YannickXiong** [comment]
@myzhan Thank you for your fellowing-up. I update the code, however still meet the same problem.
Could you help do more check? Thxs.

**Comment by myzhan** [comment]
Is locust still reporting errors? If not, are you still unable to see results on the Web-UI?

Please provider more logs, or code.

**Comment by YannickXiong** [comment]
@myzhan Locust still reported the same errors.

"/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/stats.py", line 586, in on_slave_report
global_stats.total.extend(StatsEntry.unserialize(data["stats_total"]))

**Comment by myzhan** [comment]
You need to recompile your code.

**Comment by YannickXiong** [comment]
@myzhan of course I did recompile my code. Tomorrow I will try it again. Thx.

**Comment by myzhan** [comment]
“stats_total” is still missing? That’s weird!

**Comment by YannickXiong** [comment]
@myzhan Yeas, I compare the boomer code in my pc and that one from your github, they are the same.
I recompiled my demo and re-run it, also catch fellow errors.
********deiMac:boomer *********$ locust -f dummy.py --master --master-bind-host=127.0.0.1 --master-bind-port=18090
[2017-11-22 09:03:00,046] *********deiMac.local/INFO/locust.main: Starting web monitor at *:8089
[2017-11-22 09:03:00,048] *********deiMac.local/INFO/locust.main: Starting Locust 0.8
[2017-11-22 09:03:31,532] *********deiMac.local/INFO/locust.runners: Client '*********deiMac.local_951df9e31aade3a8d4767a7df6ceb9e4' reported as ready. Currently 1 clients ready to swarm.
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: Traceback (most recent call last):
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/gevent/greenlet.py", line 536, in run
    result = self._run(*self.args, **self.kwargs)
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/runners.py", line 332, in client_listener
    events.slave_report.fire(client_id=msg.node_id, data=msg.data)
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/events.py", line 27, in fire
    handler(**kwargs)
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/locust/stats.py", line 586, in on_slave_report
    global_stats.total.extend(StatsEntry.unserialize(data["stats_total"]))
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,537] *********deiMac.local/ERROR/stderr: KeyError: 'stats_total'
[2017-11-22 09:03:34,538] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,538] *********deiMac.local/ERROR/stderr: Wed Nov 22 09:03:34 2017
[2017-11-22 09:03:34,538] *********deiMac.local/ERROR/stderr: 
[2017-11-22 09:03:34,538] *********deiMac.local/ERROR/stderr: <Greenlet at 0x10ce4f9c8: <bound method MasterLocustRunner.client_listener of <locust.runners.MasterLocustRunner object at 0x10ce80940>>> failed with KeyError

**Comment by myzhan** [comment]
I’can reproduce the problem with the latest code of both locust and boomer. Can you change boomer in your $GOPATH and print out some debug logs?

**Comment by YannickXiong** [comment]
I used tcpdump to capture all packages between locust master and the boomer client, 
and found boomer send data to master is ff:00:00:00:00:00:00:00:00:7f:03:00:4e:55:4c:4c:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00:00.  hope this helps.

because github does not support .pcap files, I changed the subfix to .pcap.pdf.
Please find the attached files. Thx.

[1.pcap.pdf](https://github.com/myzhan/boomer/files/1493602/1.pcap.pdf)

**Comment by myzhan** [comment]
I’will look into it later :D

**Comment by myzhan** [comment]
Hi @YannickXiong , I read you capture packages and extracted the No.37 package. 

![jietu20171122-192503](https://user-images.githubusercontent.com/953781/33125297-ed45ab1c-cfba-11e7-9313-b01b93cc84e3.jpg)

"stats_total" is missing, but 'xiongyangdeiMac.local_318f4e708681d604b83a4b403e202cd7' is not equals to the one in your locust log.

Is there another slave running?

**Comment by YannickXiong** [comment]
I re-tried and did deeply check on my imac, at last I found the root cause.
I found 2 boomer clones on my iMac, one is  $GOPATH/src/boomer, another is $GOPATH/src/github.com/myzan/boomer, in my demo I used the one under $GOPATH/src/boomer, while I update code for anohter one.
Now I re-correct that and finally I got the statis info on GUI. And the performance of the same script running with boomer was 11 times better than that of LOCUST, really great!

Really sorry  taking up your time just because of my mistaken.

BTW, another problem, the "Content Size" field on locust web GUI is always -1.
